grammar.js

```diff
@@ -193,7 +204,7 @@ module.exports = grammar({
     pointer_type_declarator: $ => prec.dynamic(1, prec.right(seq('*', repeat($.type_qualifier), $._type_declarator))),
     abstract_pointer_declarator: $ => prec.dynamic(1, prec.right(seq('*', repeat($.type_qualifier), optional($._abstract_declarator)))),
 
-    function_declarator: $ => prec(1, seq($._declarator, $.parameter_list)),
+    function_declarator: $ => prec(1, seq($._declarator, $.parameter_list, optional(repeat($.attribute_specifier)))),
```

Comment:
The `repeat` function already means *zero-or-more*, so you can remove the `optional` wrapper here.